server/conn.go

```diff
@@ -991,12 +986,23 @@ func (cc *clientConn) writeColumnInfo(columns []*ColumnInfo) error {
 func (cc *clientConn) writeChunks(ctx context.Context, rs ResultSet, binary bool, more bool) error {
 	data := make([]byte, 4, 1024)
 	chk := rs.NewChunk()
+	gotColumnInfo := false
 	for {
 		// Here server.tidbResultSet implements NextChunk method.
 		err := rs.NextChunk(ctx, chk)
 		if err != nil {
 			return errors.Trace(err)
 		}
+		if !gotColumnInfo {
+			// We need to call NextChunk before we get columns.
+			// Otherwise, we will get incorrect columns info.
```

Comment:
Could we move this logic to line 988?

Reply:
I think old logic is more simple. @jackysp

Reply:
I think it is the correct way to fix it. Which old logic @zz-jason ?

Reply:
Sorry, I misread the comment.

Reply:
misread +1